server/src/main/java/org/elasticsearch/common/collect/ImmutableOpenMap.java

```diff
@@ -163,6 +164,15 @@ public void remove() {
         };
     }
 
+    /**
+     * Returns a sequential unordered stream of the map entries.
+     *
+     * @return a {@link Stream} of the map entries as {@link ObjectObjectCursor}
+     */
+    public Stream<ObjectObjectCursor<KType, VType>> stream() {
```

Comment:
I think the reason this has not been added already is that it is slightly unsafe. If you look at `ObjectObjectHashMap.iterator` it explains how the `cursor` is reused throughout the iteration. I would think simple `stream().map(c -> c.value.something()).collect()` would work fine, but I am less certain of more advanced patterns. For instance `stream().sorted((a,b) -> a.value.compareTo(b.value)).something()` would not work.

I think we could devise a spliterator() that copies the elements into something like a `Map.Entry` and thereby make this safe. I would prefer to go that route to now have a safe `stream()` method.

Reply:
You're absolutely correct! Let me know if makes sense to have an internal MapEntry class which exposes `key` and `value` as public fields or use Java's Map.Entry and adapt the consumer code to use getters instead of fields!